tests/model_registry/image_validation/test_verify_rhoai_images.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from typing import Self
+from simple_logger.logger import get_logger
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.pod import Pod
+from pytest_testconfig import config as py_config
+
+from utilities.constants import DscComponents, Labels
+from utilities.general import get_csv_related_images, get_pod_images, validate_image_format
+from tests.model_registry.constants import MR_OPERATOR_NAME, MR_INSTANCE_NAME, MR_NAMESPACE
+from ocp_resources.model_registry import ModelRegistry
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.mark.parametrize(
+    "updated_dsc_component_state_scope_class",
+    [
+        {
+            "component_patch": {
+                DscComponents.MODELREGISTRY: {
+                    "managementState": DscComponents.ManagementState.MANAGED,
+                    "registriesNamespace": MR_NAMESPACE,
+                }
+            }
+        }
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("updated_dsc_component_state_scope_class")
+class TestModelRegistryImages:
+    """
+    Tests to verify that all Model Registry component images (operator and instance container images)
+    meet the requirements:
+    1. Images are hosted in registry.redhat.io
+    2. Images use sha256 digest instead of tags
+    3. Images are listed in the CSV's relatedImages section
+    """
+
+    @pytest.mark.smoke
+    def test_verify_model_registry_images(
+        self: Self, admin_client: DynamicClient, model_registry_instance: ModelRegistry
+    ):
+        # Get related images from CSV
+        related_images = get_csv_related_images(admin_client=admin_client)
+        related_image_refs = {img["image"] for img in related_images}
+
+        # Get operator pod from applications namespace
+        operator_pods = list(
+            Pod.get(
+                dyn_client=admin_client,
+                namespace=py_config["applications_namespace"],
+                label_selector=f"{Labels.OpenDataHubIo.NAME}={MR_OPERATOR_NAME}",
+            )
+        )
+        if not operator_pods:
+            pytest.fail(f"Operator pod not found in namespace {py_config['applications_namespace']}")
+        if len(operator_pods) > 1:
+            LOGGER.warning("Multiple operator pods found, using the first one")
+        operator_pod = operator_pods[0]
+
+        # Get MR instance pod(s)
+        instance_pods = list(
+            Pod.get(
+                dyn_client=admin_client,
+                namespace=MR_NAMESPACE,
+                label_selector=f"app={MR_INSTANCE_NAME}",
+            )
+        )
```

Comment:
We are testing how OLM works here.  I believe if CSV has the right images, we can assume the pods/deployments are good. Otherwise, we can write a parameterize test to check all the pods. You have quite a bit of similar code here.

Reply:
I have implemented it like this because this is the original requirement that was logged in Jira. If the requirements are updated we can always remove this logic and have a single test case for every component that validates all relatedImages in the CSV.